src/main/java/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package dao;
+
+import db.Storage;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public boolean containsKey(String key) {
+        return Storage.fruitHashMap.containsKey(key);
+    }
+
+    @Override
+    public void put(String string, Integer count) {
+        Storage.fruitHashMap.put(string, count);
+    }
+
+    @Override
+    public Set<Map.Entry<String, Integer>> getEntries() {
+        return Storage.fruitHashMap.entrySet();
```

Comment:
You don`t need entries, use map type

Reply:
I didn't understand. How is it not needed if I use it in ReportService